### DISCUSSION TITLE
New template for kiali.io

### DISCUSSION THREAD

**Discussion by lucasponce** [kiali/kiali#4293] [General]
Starting a discussion to collect feedback and opinions about the "template" (structure, not the content) of kiali.io.

Why ?

I think it has too many clicks to navigate or find what you are looking for.

What do you propose ?

Nothing specific, but that may have a side index, I see this pattern in several sites that I consider better designed:

Envoy:
![image](https://user-images.githubusercontent.com/1662329/130223902-199325d0-6545-421a-9d71-8163e425b7e5.png)

Istio:
![image](https://user-images.githubusercontent.com/1662329/130223975-a22b138b-f07e-4995-92ae-980b5675aed1.png)

Just to bring two close to the project but easily you can see this pattern more in other projects and products.

Do you want to spend too much time on this ?

No. I think we should try to keep the focus on the content, not in the templates, frameworks, or the built process of the documentation (hugo, ruby, asciidoc, etc, too much complexity there IMHO).
People can spend a ton of amount of work in a task that it should out of the scope of the project.
Within the scope is to produce content (text, images, videos), but it's out of the scope to maintain the style.
The most standard the better.

The current template is several years old but I think it's limiting in some way the way the structure of the content is added, so, before to take any decision that can be time consuming, I think it could be good to discuss about it.

Are you a webmaster from the 90s ?

No

**Comment by lucasponce** [comment]
Using the same technology for building the site, I watch one template/theme that seems to adjust to our needs [1]

[1] https://geekdocs.de/

**Comment by jmazzitelli** [comment]
I'm +1 on using a standardized template with no (or very little) required customization on our part. We should concentrate on writing content and not style. Let the template be our guide - we just fill in the content.

I'm fine with the side menu like Istio and Envoy have. Maistra also does this: https://maistra.io/docs.html

**Comment by jshaughn** [comment]
I commented on this in the "Feature" restructuring PR, but forgot to reply here.    I do like the Side Menu approach, depending on how easy it is and if there is a template.  LIke Istio, I think we could manage a landing page and perhaps some "special" pages, like "News", and then consolidate everything else under "Documentation".    +1 for content focus!

**Reply by jshaughn** [reply]
Looking at that geekdocs.de, I'm not sure I like that, it seems maybe too simple for the amount of information we have to share.

**Reply by lucasponce** [reply]
The landing page is a little bit basic, but for the content, regular pages I see it has all elements without adding new features (I mean, it has a site menu in the left, it has the page menu, it's responsive and the decorators for sections look good).
I haven't investigated too much, this one seems good and pretty similar to other sites.

What would you miss in that template ?

**Reply by lucasponce** [reply]
This one is also interesting as it focused on simplicity [1] but it has some lack of support for content pages.

[1] https://doks.netlify.app/

**Reply by leandroberetta** [reply]
I'm on the same page with @jshaughn about the landing page, although this page https://geekdocs.de/usage/getting-started/ is good and this https://doks.netlify.app/docs/prologue/introduction/ is even better.

+1 for the side menu and to have a easy template.

**Comment by israel-hdez** [comment]
I wonder if we are mainly thinking in migrating to a new template, or adjusting our current template to what other models are using?

Even if the template is old, I still like the design and I wouldn't like to see us loosing it. So, if we choose to move we may need to broadly customize the template, which may mean to circling back to maintaining the template by ourselves.

Also, if we choose to move to a completely new template, we may need to check that it is compatible with Netlify. I see the two mentioned templates require npm and hugo to build the site. Currently, we are only running hugo, so I'm not sure if Netlify will like the change. If it does not like the new tooling, we will need to find a new place to host the website, or perhaps build the website outside of Netlify and then upload the built package (which I'm not sure if it can be done).

TBH, I don't think we really need a new template. I see the main problem to solve is to minimize the number of clicks to find information, and I think this suggests a poor structure of the site. If we move to a new template keeping the same structure we won't be solving anything :-/ If a side menu is all what it takes to solve the main problem, I can volunteer to trying adding it to our current template.

**Reply by lucasponce** [reply]
The goal here is that the template should not be maintained or customized beyond the landing page.
This discussion is not about the side menu only, it's that the current template is not flexible and after some time we need to modify it.

**Reply by israel-hdez** [reply]
OK, ok... It wasn't clear from the main comment.

What kind of flexibility are you looking for?
It's to help to objectively find an appropriate template with the flexibility we need, rather than posting what I like.

**Reply by lucasponce** [reply]
Ideally I think it would be good if we find a method that the work related template (structure, style, html, css) is reduced to zero.
So, as a maintainer we should work like in a content management (i.e. medium), I mean authors should be able to just use the template and generate .md/.adoc and add images.

Today, any minor change in the structure requires to touch the template.
We can do it, we have the knowledge and skills to do it or to learn it :) I don't question that, but I think that shouldn't be the goal, we need to see the site something more static and just focus on the quality of the content.

Any template that falls into that category, to me is very useful, one of the comments in the discussion is that we could just reduce potential changes to the landing page, but rest of the documentation should be really fixed.

Another topic, but related, is that we'd need to make even easier, and perhaps to find a different way to manage versions, instead to have all versions in a branch, we could move to a branch per version, and that's all.

I know that any of the member of the team could spend a Sprint creating a new template, but giving the experience we have, I'd prefer to move to something pretty standard.

**Reply by israel-hdez** [reply]
> Ideally I think it would be good if we find a method that the work related template (structure, style, html, css) is reduced to zero.
> So, as a maintainer we should work like in a content management (i.e. medium), I mean authors should be able to just use the template and generate .md/.adoc and add images.
> 
> Today, any minor change in the structure requires to touch the template.

I wonder if you (or others) have some PRs or issues with examples of changes that we don't want to do?

Our current template is on the `themes/` folder and looking at the GitHub history, I mostly see fair changes. Content changes that required modify some part of the template have been sporadic: https://github.com/kiali/kiali.io/commits/master/themes.

If we are trying to avoid 2-3 PRs per year touching the template, I find the motivation is weak and I'm not sure if it's worth to migrate.

> Any template that falls into that category, to me is very useful

If we want to keep using Hugo, I found "[Docsy](https://themes.gohugo.io/themes/docsy/)". See a sample project here: https://example.docsy.dev/

If we are open move to something different *and* to spend a few bucks, I found "Doxy": https://themeforest.net/item/doxy-knowlegebase-documentation-jekyll-template/23209854

**Reply by lucasponce** [reply]
> I wonder if you (or others) have some PRs or issues with examples of changes that we don't want to do?

In the last weeks @jshaughn and me have do some work in the site, there are plenty of examples that modifying the structure and style is not a straightforward and it's time consuming.

Those are examples that we want to minimize or completely remove.

I don't see the motivation "weak" as now we are proposing a change of structure, including side menus for better grow a browser, and that would imply to perform a work in the template.

Yes. No one is questioning that we have skills in the team to learn and do it, but the time we invest should be used to make easier to add and update content.

The proposal here is to use something very standard and simple and just to work with that, the two of the templates that you pointed are very good.

Again, what I think we need to see the template like if we edit in a real content management, and we don't worry about this.

**Reply by israel-hdez** [reply]
> there are plenty of examples that modifying the structure and style is not a straightforward and it's time consuming.
> 

IMHO, changes in structure and styles are two fair examples that would need changes in any template :-)

> Again, what I think we need to see the template like if we edit in a real content management, and we don't worry about this.

I don't want to argue further. So, I'll give my vote of confidence that a new template is going to be good.

My only ask is to try to not lose the design that took several sprints to refine. It was a large effort and I wouldn't like to see the website getting a generic look.

FWIW, I would see OK to use one template for the "main" website, and another completely different template for the "documentation" part that suits it better.

**Reply by lucasponce** [reply]
The current design was focused on the main page and a minimal set of content, and that's good, it's a design that is active for long time.

But now the content grows in pages and also the content is more rich and we combine more screenshots, videos, code, notes, and at the moment that you navigate into a section you need to back-forth to see other sections and understand the information you want.

So, I don't see a problem trying to keep the current style for the main page but delegate the documentation in a template that offers all of these features.

**Comment by jshaughn** [comment]
I was just looking a little at Docsy and it looks like it may be promising.  I'm going to play with it a bit...
